### PR TITLE
CLI: Add "Write" to Jetpack Changelog

### PR DESCRIPTION
**Pull request by sdixon194**
#### Changes proposed in this Pull Request:

* Adds the `write` changelogger command to `jetpack changelog`

#### Jetpack product discussion

This takes a similar approach to @samiff's much simpler approach when creating the [Docker CLI](https://github.com/Automattic/jetpack/pull/19366), so if this works out, we could likely refactor the rest of the commands to work similarly. 
#### Does this pull request change what data or activity we track or use?

n/a
#### Testing instructions:

The "write" command removes all of the changelog files and adds their contents to the project's `CHANGELOG.md` file. This is destructive to the changelog files, so just don't commit any testing changes :) 

* Make sure a project has changelog files. (under projects/[project]/changelog)
* Use `jetpack changelog write` and follow the prompts to select a project.
* Upon success, check the project's CHANGELOG.md file and make sure it added the changelog file contents and removed the actual changelog files from the changelog folder.  
* It accepts passing a project directly, ie `jetpack changelog write plugins/jetpack`

It also accepts a lot of passthrough arguments, but truthfully I wasn't too sure how to test some of them ([--use-version USE-VERSION] [--use-significance USE-SIGNIFICANCE]), so any help with testing that would be cool. The ones I could figure out seemed to work fine.

### CODE REVIEW

File: tools/cli/commands/changelog.js [hunk 1]
```diff
@@ -99,6 +99,28 @@ export function changelogDefine( yargs ) {
...
+								type: 'string',
+							} )
+							.option( 'dev', {
+								describe: 'Output validation issues using GitHub Action command syntax.',
+								type: 'bool',
+							} )
+							.option( 'prod', {
+								describe: 'Output file paths in this directory relative to it.',
+								type: 'bool',
+							} );
```
Comment (anomiex) [inline tools/cli/commands/changelog.js:119]: Neither of these are valid options to `changelogger write`.

File: tools/cli/commands/changelog.js [hunk 2]
```diff
@@ -221,6 +243,20 @@ export async function changelogValidate( argv ) {
...
 }
 
+/**
+ * Changelog write script.
+ *
+ * @param {object} argv - arguments passed to the CLI.
+ */
+async function changelogWrite( argv ) {
+	argv = await validateProject( argv );
+	const args = process.argv.slice( 5 );
```
Comment (anomiex) [inline tools/cli/commands/changelog.js:253]: I note this fails oddly if you put args before the project name, e.g. `jetpack changelog write --use-version=100.0.0 plugins/jetpack`

**Review by anomiex — COMMENTED** [review]
> It also accepts a lot of passthrough arguments, but truthfully I wasn't too sure how to test some of them ([--use-version USE-VERSION] [--use-significance USE-SIGNIFICANCE]), so any help with testing that would be cool. The ones I could figure out seemed to work fine.

`--use-version` is pretty simple: Use it to specify a version number, then see if that version number shows up in CHANGELOG.md instead of whatever it would normally use.

To test `--use-significance`, you might do something like `jetpack changelog write packages/changelogger --use-significance=major` then see that CHANGELOG.md has the new version as 2.0.0 instead of 1.1.2.

Both of those seemed to work.

File: tools/cli/commands/changelog.js (hunk 1 above)
Comment (sdixon194) [reply tools/cli/commands/changelog.js:119]: Oh nice catch, I intended to fix those.

**Comment by sdixon194** [comment]
> To test `--use-significance`, you might do something like `jetpack changelog write packages/changelogger --use-significance=major` then see that CHANGELOG.md has the new version as 2.0.0 instead of 1.1.2.

Ahh okay, I think the issue was I wasn't formatting the version correctly (2.0 vs 2.0.0, etc, didn't know the last point was required.

File: tools/cli/commands/changelog.js (hunk 2 above)
Comment (sdixon194) [reply tools/cli/commands/changelog.js]: Redid this a bit. Should be fixed in latest commit.

**Review by samiff — COMMENTED** [review]
Looking really nice 👍 

- `jetpack changelog write --help` ✅
- `jetpack changelog` ✅
- `jetpack changelog write` ✅
- `jetpack changelog write plugins/jetpack` ✅
- `jetpack changelog write plugins/jetpack --use-version=10.0` ✅
- `jetpack changelog write --use-version=10.0 plugins/jetpack` ✅
- `jetpack changelog write --use-version=10.0` ✅
- `jetpack changelog write --use-version=10.1 plugins/jetpack --release-date="unreleased"` ✅
- `jetpack changelog --use-version=10.0` ❌ 
  - this falls outside of the `jetpack changelog <write>` format, so wouldn't really expect it to work.
- `jetpack changelog write plugins/jetpack --use-significance=major` ❌
  - or `composer exec -- changelogger write --use-significance=major` only bumped from 9.6 --> 9.7 but I guess this is a "special" case since Jetpack doesn't have the patch piece of semver @anomiex  ? The following worked just fine for example:
  - `jetpack changelog write packages/connection-ui --use-significance=major`

**Comment by kraftbj** [comment]
>     * but I guess this is a "special" case since Jetpack doesn't have the patch piece of semver

Correct. The Jetpack plugin uses the "wordpress" scheme, so changelog doesn't try to implement semver: https://github.com/Automattic/jetpack/blob/master/projects/plugins/jetpack/composer.json#L105

**Review by kraftbj — APPROVED** [review]

**Pull request merged**